app/models/solidus_bolt/gateway.rb

```diff
@@ -51,8 +51,22 @@ def credit(_amount, _response_code, _gateway_options)
       raise NotImplementedError, 'credit method has not been implemented in SolidusBolt::Gateway class'
     end
 
-    def purchase(_float_amount, _response_code, _gateway_options)
-      raise NotImplementedError, 'purchase method has not been implemented in SolidusBolt::Gateway class'
+    def purchase(_float_amount, payment_source, gateway_options)
+      order_number = gateway_options[:order_id].split('-').first
+      order = Spree::Order.find_by(number: order_number)
```

Comment:
since we are repeating this code in several methods, how about extracting it to a private method?

Reply:
done
check here [54fd960](https://github.com/nebulab/solidus_bolt/commit/54fd9608da840e0888caedbe9310311acda1e891)

Reply:
I think we should pass the line above too